src/timelock.cairo

```diff
@@ -14,23 +18,119 @@ trait ITimelock<TStorage> {
     fn execute(ref self: TStorage, calls: Span<Call>) -> Array<Span<felt252>>;
 
     // Return the execution window, i.e. the start and end timestamp in which the call can be executed
-    fn get_execution_window(self: @TStorage, id: felt252) -> (u64, u64);
+    fn get_execution_window(self: @TStorage, id: felt252) -> ExecutionWindow;
 
     // Get the current owner
     fn get_owner(self: @TStorage) -> ContractAddress;
 
     // Returns the delay and the window for call execution
-    fn get_configuration(self: @TStorage) -> (u64, u64);
+    fn get_configuration(self: @TStorage) -> TimelockConfig;
 
     // Transfer ownership, i.e. the address that can queue and cancel calls. This must be self-called via #queue.
     fn transfer(ref self: TStorage, to: ContractAddress);
     // Configure the delay and the window for call execution. This must be self-called via #queue.
-    fn configure(ref self: TStorage, delay: u64, window: u64);
+    fn configure(ref self: TStorage, delay_and_window: TimelockConfig);
 }
 
+
+const U64_MASK: u128 = 0xFFFFFFFFFFFFFFFF;
+const TWO_POW_64: u128 = 0x10000000000000000;
+
+impl ThreeU64TupleStorePacking of StorePacking<(u64, u64, u64), felt252> {
```

Comment:
do these both belong in the timelock file?

Reply:
I can create a file dedicated to the timestamp optimization